src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java

```diff
@@ -314,21 +286,10 @@ public void setBuildingLevel(final int level)
         return new ArrayList<>(bedList);
     }
 
-    /**
-     * Returns whether the citizen has this as home or not.
-     *
-     * @param citizen Citizen to check.
-     * @return True if citizen lives here, otherwise false.
-     */
-    public boolean hasResident(final CitizenData citizen)
-    {
-        return residents.contains(citizen);
-    }
-
     /**
      * The view of the citizen hut.
      */
-    public static class View extends AbstractBuildingHut.View
+    public static class View extends AbstractBuildingView
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 38 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)